scripts/coverage_diff.py

```diff
@@ -107,9 +105,28 @@ def coverage_table(coverage, format='txt'):
     """
     columns = ["File Name", "Coverage (%)", "Delta Coverage"]
     separator = ['-'*len(cn) for cn in columns]
-    result = [columns, separator] + [[fn, coverage[fn]["child_cov"], coverage[fn]["delta_cov"]] for fn in coverage]
+    result = [columns, separator] + [[fn, coverage[fn]['child_cov'], coverage[fn]['delta_cov']] for fn in coverage]
     return txt_table(result) if format == 'txt' else markdown_table(result)
 
+def file_coverage_status(fn_coverage):
+    if fn_coverage["new_file"] and fn_coverage["delta_cov"] <= 20:
+        # New files need to have at least 20% coverage
+        return {"exit_code":1, "message":"New files need at least 20% coverage.", "log_level":"error", "terminate":True} 
+    elif fn_coverage["delta_cov"] < 0 and abs(fn_coverage["delta_cov"]) >= min(20, fn_coverage["parent_cov"]):
+        return {"exit_code":1, "message":"Coverage decreased severely for more than 20% or a file lost coverage.", "log_level":"error", "terminate":True} 
+    elif fn_coverage["delta_cov"] < 0:
+        return {"exit_code":0, "message":"Coverage decreased slightly for less than 20%", "log_level":"warning", "terminate":True} 
+    else:
+        return {"exit_code":0, "message":"Coverage non decreasing.", "log_level":"notice", "terminate":False}
+    
+def annotate_files(coverage):
+    """Return the exit code and exit message for given coverage
+    coverage: {filename: {parent_cov: float, child_cov: float, delta_cov: float, new_file: bool}}
+    output: None
+    """
+    
+    [print(f"::{file_coverage_status(coverage[fn])['log_level']} file={fn},line=1,endLine=1,title=Coverage::{coverage[fn]['child_cov']}% ({coverage[fn]['delta_cov']}%)") for fn in coverage if abs(coverage[fn]['delta_cov']) > 1]
```

Comment:
You can expand this loop to make it much easier to comprehend:
```python
for fn, cov in coverage.items():
  if not abs(cov['delta_cov']) > 1:
    continue

  status = file_coverage_status(cov)
  print(f"::{status['log_level']} file={fn},line=1,endLine=1,title=Coverage::{cov['child_cov']}% ({cov['delta_cov']}%)")
```
The idea is to write your code vertically, not horizontally.